x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_details_page/components/agent_diagnostics/index.tsx

```diff
@@ -0,0 +1,218 @@
+
+  const columns: Array<EuiTableFieldDataColumnType<AgentDiagnostics>> = [
+    {
+      field: 'id',
+      name: 'File',
+      render: (id: string) => {
+        const currentItem = diagnosticsEntries.find((item) => item.id === id);
+        return currentItem?.status === 'READY' ? (
+          <EuiLink href={getAbsolutePath(currentItem?.filePath)} download target="_blank">
+            <EuiIcon type="download" /> &nbsp; {currentItem?.name}
+          </EuiLink>
+        ) : currentItem?.status === 'IN_PROGRESS' || currentItem?.status === 'AWAITING_UPLOAD' ? (
+          <EuiText color="subdued">
+            <EuiLoadingSpinner /> &nbsp;
+            <FormattedMessage
+              id="xpack.fleet.requestDiagnostics.generatingText"
+              defaultMessage="Generating diagnostics file..."
+            />
+          </EuiText>
+        ) : (
+          <EuiText color="subdued">
+            <EuiIcon type="alert" color="red" /> &nbsp;
+            {currentItem?.name}
+          </EuiText>
+        );
+      },
+    },
+    {
+      field: 'id',
+      name: 'Date',
+      dataType: 'date',
+      render: (id: string) => {
+        const currentItem = diagnosticsEntries.find((item) => item.id === id);
+        return (
+          <EuiText color={currentItem?.status === 'READY' ? 'default' : 'subdued'}>
+            {formatDate(currentItem?.createTime, 'll')}
+          </EuiText>
+        );
+      },
+    },
+  ];
+
+  async function onSubmit() {
+    try {
+      setIsSubmitting(true);
+      const { error } = await sendPostRequestDiagnostics(agent.id);
+      if (error) {
+        throw error;
+      }
+      setIsSubmitting(false);
+      const successMessage = i18n.translate(
+        'xpack.fleet.requestDiagnostics.successSingleNotificationTitle',
+        {
+          defaultMessage: 'Request diagnostics submitted',
+        }
+      );
+      notifications.toasts.addSuccess(successMessage);
+    } catch (error) {
+      setIsSubmitting(false);
+      notifications.toasts.addError(error, {
+        title: i18n.translate('xpack.fleet.requestDiagnostics.fatalErrorNotificationTitle', {
+          defaultMessage:
+            'Error requesting diagnostics {count, plural, one {agent} other {agents}}',
+          values: { count: 1 },
+        }),
+      });
+    }
+  }
+
+  return (
+    <EuiFlexGroup direction="column" gutterSize="l">
+      <EuiFlexItem>
+        <EuiCallOut
+          iconType="alert"
+          color="warning"
+          title={
+            <FormattedMessage
+              id="xpack.fleet.fleetServerSetup.calloutTitle"
+              defaultMessage="Agent diagnostics"
+            />
+          }
+        >
+          <FormattedMessage
+            id="xpack.fleet.requestDiagnostics.calloutText"
+            defaultMessage="Diagnostics files are stored in Elasticsearch, and as such can incur storage costs. Fleet will automatically remove old diagnostics files after 30 days."
```

Comment:
This callout text might have to be updated if we agreed on a different max age then 30 days.